ccl/backupccl: TestBackupAndRestoreJobDescription failed

ccl/backupccl.TestBackupAndRestoreJobDescription [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11078538?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11078538?buildTab=artifacts#/) on release-23.1 @ [aaedb3dd0d97097cff7afeb2532c6265d3d62818](https://github.com/cockroachdb/cockroach/commits/aaedb3dd0d97097cff7afeb2532c6265d3d62818):


```
=== RUN   TestBackupAndRestoreJobDescription
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/95e138e66d69292427dfb9528cf06d04/logTestBackupAndRestoreJobDescription332439570
    test_log_scope.go:79: use -show-logs to present logs inline
    backup_test.go:882: query 'SELECT description FROM [SHOW JOBS] WHERE job_type='RESTORE'': expected:
        RESTORE DATABASE data FROM ('nodelocal://1/foo/1?COCKROACH_LOCALITY=default', 'nodelocal://1/foo/2?COCKROACH_LOCALITY=dc%3Ddc1', 'nodelocal://1/foo/3?COCKROACH_LOCALITY=dc%3Ddc2')
        RESTORE DATABASE data FROM ('nodelocal://1/full/2023/07/27-052854.36?COCKROACH_LOCALITY=default', 'nodelocal://2/full/2023/07/27-052854.36?COCKROACH_LOCALITY=dc%3Ddc1', 'nodelocal://3/full/2023/07/27-052854.36?COCKROACH_LOCALITY=dc%3Ddc2')
        RESTORE DATABASE data FROM ('nodelocal://1/full/2023/07/27-052854.36?COCKROACH_LOCALITY=default', 'nodelocal://2/full/2023/07/27-052854.36?COCKROACH_LOCALITY=dc%3Ddc1', 'nodelocal://3/full/2023/07/27-052854.36?COCKROACH_LOCALITY=dc%3Ddc2') WITH incremental_location = ('nodelocal://1/inc/2023/07/27-052854.36?COCKROACH_LOCALITY=default', 'nodelocal://3/inc/2023/07/27-052854.36?COCKROACH_LOCALITY=dc%3Ddc1', 'nodelocal://3/inc/2023/07/27-052854.36?COCKROACH_LOCALITY=dc%3Ddc2')
        RESTORE DATABASE data FROM ('nodelocal://1/full/2023/07/27-052857.20?COCKROACH_LOCALITY=default', 'nodelocal://2/full/2023/07/27-052857.20?COCKROACH_LOCALITY=dc%3Ddc1', 'nodelocal://3/full/2023/07/27-052857.20?COCKROACH_LOCALITY=dc%3Ddc2')
        RESTORE DATABASE data FROM ('nodelocal://1/full/2023/07/27-052857.20?COCKROACH_LOCALITY=default', 'nodelocal://2/full/2023/07/27-052857.20?COCKROACH_LOCALITY=dc%3Ddc1', 'nodelocal://3/full/2023/07/27-052857.20?COCKROACH_LOCALITY=dc%3Ddc2')
        
        got:
        RESTORE DATABASE data FROM ('nodelocal://1/foo/1?COCKROACH_LOCALITY=default', 'nodelocal://1/foo/2?COCKROACH_LOCALITY=dc%3Ddc1', 'nodelocal://1/foo/3?COCKROACH_LOCALITY=dc%3Ddc2')
        RESTORE DATABASE data FROM ('nodelocal://1/full/2023/07/27-052857.20?COCKROACH_LOCALITY=default', 'nodelocal://2/full/2023/07/27-052857.20?COCKROACH_LOCALITY=dc%3Ddc1', 'nodelocal://3/full/2023/07/27-052857.20?COCKROACH_LOCALITY=dc%3Ddc2')
        RESTORE DATABASE data FROM ('nodelocal://1/full/2023/07/27-052854.36?COCKROACH_LOCALITY=default', 'nodelocal://2/full/2023/07/27-052854.36?COCKROACH_LOCALITY=dc%3Ddc1', 'nodelocal://3/full/2023/07/27-052854.36?COCKROACH_LOCALITY=dc%3Ddc2')
        RESTORE DATABASE data FROM ('nodelocal://1/full/2023/07/27-052854.36?COCKROACH_LOCALITY=default', 'nodelocal://2/full/2023/07/27-052854.36?COCKROACH_LOCALITY=dc%3Ddc1', 'nodelocal://3/full/2023/07/27-052854.36?COCKROACH_LOCALITY=dc%3Ddc2') WITH incremental_location = ('nodelocal://1/inc/2023/07/27-052854.36?COCKROACH_LOCALITY=default', 'nodelocal://3/inc/2023/07/27-052854.36?COCKROACH_LOCALITY=dc%3Ddc1', 'nodelocal://3/inc/2023/07/27-052854.36?COCKROACH_LOCALITY=dc%3Ddc2')
        RESTORE DATABASE data FROM ('nodelocal://1/full/2023/07/27-052857.20?COCKROACH_LOCALITY=default', 'nodelocal://2/full/2023/07/27-052857.20?COCKROACH_LOCALITY=dc%3Ddc1', 'nodelocal://3/full/2023/07/27-052857.20?COCKROACH_LOCALITY=dc%3Ddc2')
        
    testutils.go:199: no Invalid Descriptors
    panic.go:540: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/95e138e66d69292427dfb9528cf06d04/logTestBackupAndRestoreJobDescription332439570
--- FAIL: TestBackupAndRestoreJobDescription (66.77s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackupAndRestoreJobDescription.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-30157

--- a/pkg/ccl/backupccl/backup_test.go
+++ b/pkg/ccl/backupccl/backup_test.go
@@ -897,7 +897,7 @@ func TestBackupAndRestoreJobDescription(t *testing.T) {
 	resolvedAsOfCollectionURIs := getResolvedCollectionURIs(collections, asOf1)
 
 	sqlDB.CheckQueryResults(
-		t, "SELECT description FROM [SHOW JOBS] WHERE job_type='RESTORE'",
+		t, "SELECT description FROM [SHOW JOBS] WHERE job_type='RESTORE' ORDER BY created",
 		[][]string{
 			{fmt.Sprintf("RESTORE DATABASE data FROM ('%s', '%s', '%s')",
 				backups[0].(string), backups[1].(string), backups[2].(string))},

